### PR TITLE
Closing previous PR and creating a new one.

### PR DESCRIPTION
**Pull request by bamartinezd**
Includes Training algorithms and Amazon assessment.

### CODE REVIEW

File: Amazon Transaction Logs/Program.cs [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+            string logsItem = Console.ReadLine();
+            logs.Add(logsItem);
+        }
+
+        int threshold = Convert.ToInt32(Console.ReadLine().Trim());
+
+        List<string> result = processLogs(logs, threshold);
+    }
+
+    public static List<string> processLogs(List<string> logs, int threshold)
```
Comment (andrescollazosc) [inline Amazon Transaction Logs/Program.cs]: - This Method is not working, Please remove it.
 - Please read the documentation Code style to c# (https://google.github.io/styleguide/csharp-style.html).

File: Amazon Transaction Logs/Program.cs [hunk 2]
```diff
@@ -0,0 +1,24 @@
+﻿public class Program
+{
+    public static void Main(string[] args)
+    {
+        int logsCount = Convert.ToInt32(Console.ReadLine().Trim());
+
+        List<string> logs = new List<string>();
```
Comment (andrescollazosc) [inline Amazon Transaction Logs/Program.cs]: Replace List<string> logs = new List<string>(); By var logs = new List<string>();

File: AmazonDeliveryCenters/Program.cs [hunk 3]
```diff
@@ -0,0 +1,35 @@
+﻿using System.Text.Json;
+
+public class Program
+{
+    public static void Main(string[] args)
+    {
+        //var a = suitableLocations(new List<int> { 2, 0, 3, -4 }, 22);
+        //var a = suitableLocations(new List<int> { -2, 1, 0 }, 8);
```
Comment (andrescollazosc) [inline AmazonDeliveryCenters/Program.cs]: Remove Code that is Not being used.

File: AmazonDeliveryCenters/Program.cs [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+        var a = suitableLocations(new List<int> { -1, 0, 1 }, 8);
```
Comment (andrescollazosc) [inline AmazonDeliveryCenters/Program.cs]: Please, Read Docs C# Style ([List<string> logs = new List<string>();](https://google.github.io/styleguide/csharp-style.html)), changed a By a descriptive word.

File: AmazonTransactionLogs/Program.cs [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+
+public class Program
+{
+    public static void Main(string[] args)
+    {
+        var logs = new List<string> { "1 2 50", "1 7 70", "1 3 20", "2 2 17" };
+
+        int threshold = 2;
+
+        List<string> result = processLogs(logs, threshold);
```
Comment (andrescollazosc) [inline AmazonTransactionLogs/Program.cs]: - please replace List<string> By var.

File: AmazonTransactionLogs/Program.cs [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+
+//https://www.hackerrank.com/test/63ek10mhil5/60d306ab105867d80fca544041154273
+
+public class Program
+{
+    public static void Main(string[] args)
+    {
+        var logs = new List<string> { "1 2 50", "1 7 70", "1 3 20", "2 2 17" };
+
+        int threshold = 2;
```
Comment (andrescollazosc) [inline AmazonTransactionLogs/Program.cs]: Please add const int threshold = 2;

File: AmazonTransactionLogs/Program.cs [hunk 7]
```diff
@@ -0,0 +1,62 @@
...
+
+        List<string> result = processLogs(logs, threshold);
+    }
+
+    public static List<string> processLogs(List<string> logs, int threshold)
```
Comment (andrescollazosc) [inline AmazonTransactionLogs/Program.cs]: Please change Name processLogs By ProcessLogs. Read documentation(https://google.github.io/styleguide/csharp-style.html)

File: AmazonTransactionLogs/Program.cs [hunk 8]
```diff
@@ -0,0 +1,62 @@
...
+            }
+            else
+            {
+                resultIds.Add(sender_user_id);
+                resultIds.Add(recipient_user_id);
+            }
+
+            
+        });
+
```
Comment (andrescollazosc) [inline AmazonTransactionLogs/Program.cs]: Please refactor or split this code in another method.

File: AmazonTransactionLogs/Program.cs [hunk 9]
```diff
@@ -0,0 +1,62 @@
...
+        var a = resultIds.GroupBy(id => id)
```
Comment (andrescollazosc) [inline AmazonTransactionLogs/Program.cs]: Replace character a By Descriptive Word

File: AmazonTransactionLogs/Program.cs [hunk 10]
```diff
@@ -0,0 +1,62 @@
...
+                {
+                    Count = g.Count(),
+                    id = g.Key,
+                };
+            });
+
+        List<string> finalIds = new List<string>();
+        foreach (var keyValue in a)
+        {
+            if (keyValue.Count>=threshold)
```
Comment (andrescollazosc) [inline AmazonTransactionLogs/Program.cs]: Please Read doc (https://google.github.io/styleguide/csharp-style.html)

File: BookSearchPAge/Program.cs [hunk 11]
```diff
@@ -0,0 +1,47 @@
+﻿public class Program
+{
+    public static void Main(string[] args)
+    {
+        var a = SearchPage(6, 4);
```
Comment (andrescollazosc) [inline BookSearchPAge/Program.cs:5]: - Variable a is not used.
 - Replace character a by a descriptive word.

File: BookSearchPAge/Program.cs [hunk 12]
```diff
@@ -0,0 +1,47 @@
+﻿public class Program
+{
+    public static void Main(string[] args)
+    {
+        var a = SearchPage(6, 4);
+    }
+
+    public int SearchPage(int n, int p)
```
Comment (andrescollazosc) [inline BookSearchPAge/Program.cs]: Please check all this method.

File: BookSearchPAge/Program.cs [hunk 13]
```diff
@@ -0,0 +1,47 @@
...
+        
+    }
+
+    private static bool IsPair(int n){
+        return n % 2 == 0;
+    }
+}
+
+class Page
+{
```
Comment (andrescollazosc) [inline BookSearchPAge/Program.cs:44]: Please separate this class in another file

File: BruteForceAlgorithm/Program.cs [hunk 14]
```diff
@@ -0,0 +1,66 @@
...
+public class Program
+{
+    public delegate bool DelBruteForceAlgo(ref char[] inputs);
+    public static void Main(String[] args) {
+        var watch = new Stopwatch();
+        int tries = 0;
+        watch.Start();
+
+        DelBruteForceAlgo testCallback = delegate (ref char[] inputs) {
+            var str = new String(inputs);
```
Comment (andrescollazosc) [inline BruteForceAlgorithm/Program.cs:12]: Please changed variable name str by a descriptive name.

File: BruteForceAlgorithm/Program.cs [hunk 15]
```diff
@@ -0,0 +1,66 @@
...
+            Console.WriteLine($"testCallback - {str}");
+            tries++;
+            return (str == "f71d05c0");
```
Comment (andrescollazosc) [inline BruteForceAlgorithm/Program.cs:15]: Please include this value 'f71d05c0'  in a constant variable

File: BruteForceAlgorithm/Program.cs [hunk 16]
```diff
@@ -0,0 +1,66 @@
...
+        };
+
+        bool result = BruteForce("01234567abcdef", 1, 8, testCallback);
```
Comment (andrescollazosc) [inline BruteForceAlgorithm/Program.cs:18]: Please check result variable, Is not used it.

File: ConsoleApp_fit_point74/Program.cs [hunk 17]
```diff
@@ -0,0 +1,31 @@
+﻿using System;
+
+namespace ConsoleApp_fit_point74
```
Comment (andrescollazosc) [inline ConsoleApp_fit_point74/Program.cs:3]: Please read documentation

File: ConsoleApp_fit_point74/Program.cs [hunk 18]
```diff
@@ -0,0 +1,31 @@
...
+            for (int i = 0; i < arrayToBeReversed.Length; i++)
+            {
+                reversed[invertedIndex] = arrayToBeReversed[i];
+                invertedIndex--;
+            }
+
+            return reversed;
+        }
+    }
+}
```
Comment (andrescollazosc) [inline ConsoleApp_fit_point74/Program.cs:31]: LGTM

File: DivisibleSumPairs/Program.cs [hunk 19]
```diff
@@ -0,0 +1,24 @@
+﻿public class Program
+{
+    public static void Main(string[] args) {
+        var res = divisibleSumPairs(6, 5, new List<int> { 1, 2, 3, 4, 5, 6 });
+        Console.WriteLine(res);
+    }
+
+    private static int divisibleSumPairs(int n, int k, List<int> ar)
```
Comment (andrescollazosc) [inline DivisibleSumPairs/Program.cs:8]: Please read documentation

File: DivisibleSumPairs/Program.cs [hunk 20]
```diff
@@ -0,0 +1,24 @@
...
+    {
+        int pairs = 0;
+        
+        for (int i = 0; i < ar.Count; i++)
+        {
+            for (int j = i + 1; j < ar.Count; j++)
+            {
+                int res = ar[i] + ar[j];
+                if (res % k == 0)
```
Comment (andrescollazosc) [inline DivisibleSumPairs/Program.cs:17]: Pleas read documentation.

File: DivisibleSumPairs/Program.cs (hunk 19 above)
Comment (andrescollazosc) [inline DivisibleSumPairs/Program.cs:8]: define variable names

File: Amazon Transaction Logs/Program.cs (hunk 1 above)
Comment (bamartinezd) [reply Amazon Transaction Logs/Program.cs]: Resolved, removing the duplicated project from the solution.

File: Amazon Transaction Logs/Program.cs (hunk 2 above)
Comment (bamartinezd) [reply Amazon Transaction Logs/Program.cs]: Resolved, removing the duplicated project from the solution.

File: AmazonDeliveryCenters/Program.cs (hunk 3 above)
Comment (bamartinezd) [reply AmazonDeliveryCenters/Program.cs]: So far, we need them as possible scenarios in order to test this method, but we are considering creating unit tests.

File: AmazonDeliveryCenters/Program.cs (hunk 4 above)
Comment (bamartinezd) [reply AmazonDeliveryCenters/Program.cs]: Resolved, replacing the a by suitableCenters.

File: AmazonTransactionLogs/Program.cs (hunk 5 above)
Comment (bamartinezd) [reply AmazonTransactionLogs/Program.cs]: .l. 
https://github.com/dotnet/runtime/blob/main/docs/coding-guidelines/coding-style.md

10. We only use var when the type is explicitly named on the right-hand side...

**Pull request merged**